### ISSUE TITLE
Difficulty setting dimensions while distributing subviews

### ISSUE THREAD

**Issue by rboisjoly** [smileyborg/UIView-AutoLayout#3]
I'm trying to predictably set a UIView's width, and then distribute subviews within. 

It seems if I try to do both, I get an irreconciliable condition.

Here is an example:

```
    [thisWeekView autoSetDimension:ALDimensionWidth toSize:someWidthValue];
    [thisWeekView autoDistributeSubviews:[thisWeekView subviews] alongAxis:ALAxisHorizontal withFixedSpacing:1 alignment:NSLayoutFormatAlignAllCenterY];
```

I then get:

Unable to simultaneously satisfy constraints...

Here is one listing of the constraints it is showing:
"<NSLayoutConstraint:0x102b2190 H:[UILabel:0x102b1f80(19.2)]>",
    "<NSLayoutConstraint:0x102b3220 H:[UIView:0x102aa270(175)]>",
    "<NSLayoutConstraint:0x102b32e0 H:|-(1)-[UILabel:0x102b1f80]   (Names: '|':UIView:0x102aa270 )>",
    "<NSLayoutConstraint:0x102b3320 H:[UILabel:0x102b1f80]-(1)-[UILabel:0x102b2200]>",
    "<NSLayoutConstraint:0x102b1cb0 H:[UILabel:0x102b2200]-(1)-[UILabel:0x102b24b0]>",
    "<NSLayoutConstraint:0x102b33f0 UILabel:0x102b24b0.width == UILabel:0x102b2200.width>",
    "<NSLayoutConstraint:0x102b3450 H:[UILabel:0x102b24b0]-(1)-[UILabel:0x102b2760]>",
    "<NSLayoutConstraint:0x102b34c0 UILabel:0x102b2760.width == UILabel:0x102b24b0.width>",
    "<NSLayoutConstraint:0x102b3560 H:[UILabel:0x102b2760]-(1)-[UILabel:0x102b2a10]>",
    "<NSLayoutConstraint:0x102b3590 UILabel:0x102b2a10.width == UILabel:0x102b2760.width>",
    "<NSLayoutConstraint:0x102b35f0 H:[UILabel:0x102b2a10]-(1)-[UILabel:0x102b2cc0]>",
    "<NSLayoutConstraint:0x102b3620 UILabel:0x102b2cc0.width == UILabel:0x102b2a10.width>",
    "<NSLayoutConstraint:0x102b33b0 H:[UILabel:0x102b2cc0]-(1)-[UILabel:0x102b2f70]>",
    "<NSLayoutConstraint:0x102b3480 UILabel:0x102b2f70.width == UILabel:0x102b2cc0.width>",
    "<NSLayoutConstraint:0x102b3790 UILabel:0x102b2f70.trailing == UIView:0x102aa270.trailing - 1>",
    "<NSLayoutConstraint:0x102b3350 UILabel:0x102b2200.width == UILabel:0x102b1f80.width>"

Basically, I have a UIView containing 7 subviews which are UILabels. I need to distribute these across the main UIView and have this main UIView a fixed size.

maybe I'm just doing this wrong...

**Comment by rboisjoly** [comment]
great... not showing the actual constraints...

**Comment by rboisjoly** [comment]
"NSLayoutConstraint:0x102a82e0 H:[UILabel:0x102a8120(19)]",
    "NSLayoutConstraint:0x102a99d0 H:|-(1)-[UILabel:0x102a8120](Names: '|':UIView:0x102a8040)",
    "NSLayoutConstraint:0x102a9a40 H:[UILabel:0x102a8120]-(1)-[UILabel:0x102a8cc0]",
    "NSLayoutConstraint:0x102a9ad0 H:[UILabel:0x102a8cc0]-(1)-[UILabel:0x102a8ed0]",
    "NSLayoutConstraint:0x102a9b40 UILabel:0x102a8ed0.width == UILabel:0x102a8cc0.width",
    "NSLayoutConstraint:0x102a9ba0 H:[UILabel:0x102a8ed0]-(1)-[UILabel:0x102a90f0]",
    "NSLayoutConstraint:0x102a9bd0 UILabel:0x102a90f0.width == UILabel:0x102a8ed0.width",
    "NSLayoutConstraint:0x102a9cb0 H:[UILabel:0x102a90f0]-(1)-[UILabel:0x102a9310]",
    "NSLayoutConstraint:0x102a9ce0 UILabel:0x102a9310.width == UILabel:0x102a90f0.width",
    "NSLayoutConstraint:0x102a9d40 H:[UILabel:0x102a9310]-(1)-[UILabel:0x102a9530]",
    "NSLayoutConstraint:0x102a9d70 UILabel:0x102a9530.width == UILabel:0x102a9310.width",
    "NSLayoutConstraint:0x102a9dd0 H:[UILabel:0x102a9530]-(1)-[UILabel:0x102a9750]",
    "NSLayoutConstraint:0x102a9c00 UILabel:0x102a9750.width == UILabel:0x102a9530.width",
    "NSLayoutConstraint:0x102a9ee0 UILabel:0x102a9750.trailing == UIView:0x102a8040.trailing - 1",
    "NSLayoutConstraint:0x102aa440 H:[UIView:0x102a8040(175)]",
    "NSLayoutConstraint:0x102a9a70 UILabel:0x102a8cc0.width == UILabel:0x102a8120.width"

**Comment by smileyborg** [comment]
That looks correct and it should work fine (as long as the width of the containing thisWeekView is wide enough to allow each label some width that is >0.

Can you copy paste the entire exception from the debug console?

**Comment by rboisjoly** [comment]
In case you are wondering, trying to build a UIView for a week, where each day has a fixed width and distributed within the week view.

I also need a set of 7 "Header" views in a separate UIViews to place the day Initial.

Things are not aligning easily...

As for Copy/Paste, here you go:

2013-09-07 16:19:42.332 AgendaScolaire[13737:a0b] Unable to simultaneously satisfy constraints.
    Probably at least one of the constraints in the following list is one you don't want. Try this: (1) look at each constraint and try to figure out which you don't expect; (2) find the code that added the unwanted constraint or constraints and fix it. (Note: If you're seeing NSAutoresizingMaskLayoutConstraints that you don't understand, refer to the documentation for the UIView property translatesAutoresizingMaskIntoConstraints) 
(
    "«NSLayoutConstraint:0x102a82e0 H:[UILabel:0x102a8120(19)]»",
    "«NSLayoutConstraint:0x102a99d0 H:|-(1)-[UILabel:0x102a8120](Names: '|':UIView:0x102a8040)»",
    "«NSLayoutConstraint:0x102a9a40 H:[UILabel:0x102a8120]-(1)-[UILabel:0x102a8cc0]»",
    "«NSLayoutConstraint:0x102a9ad0 H:[UILabel:0x102a8cc0]-(1)-[UILabel:0x102a8ed0]»",
    "«NSLayoutConstraint:0x102a9b40 UILabel:0x102a8ed0.width == UILabel:0x102a8cc0.width»",
    "«NSLayoutConstraint:0x102a9ba0 H:[UILabel:0x102a8ed0]-(1)-[UILabel:0x102a90f0]»",
    "«NSLayoutConstraint:0x102a9bd0 UILabel:0x102a90f0.width == UILabel:0x102a8ed0.width»",
    "«NSLayoutConstraint:0x102a9cb0 H:[UILabel:0x102a90f0]-(1)-[UILabel:0x102a9310]»",
    "«NSLayoutConstraint:0x102a9ce0 UILabel:0x102a9310.width == UILabel:0x102a90f0.width»",
    "«NSLayoutConstraint:0x102a9d40 H:[UILabel:0x102a9310]-(1)-[UILabel:0x102a9530]»",
    "«NSLayoutConstraint:0x102a9d70 UILabel:0x102a9530.width == UILabel:0x102a9310.width»",
    "«NSLayoutConstraint:0x102a9dd0 H:[UILabel:0x102a9530]-(1)-[UILabel:0x102a9750]»",
    "«NSLayoutConstraint:0x102a9c00 UILabel:0x102a9750.width == UILabel:0x102a9530.width»",
    "«NSLayoutConstraint:0x102a9ee0 UILabel:0x102a9750.trailing == UIView:0x102a8040.trailing - 1»",
    "«NSLayoutConstraint:0x102aa440 H:[UIView:0x102a8040(175)]»",
    "«NSLayoutConstraint:0x102a9a70 UILabel:0x102a8cc0.width == UILabel:0x102a8120.width»"
)

Will attempt to recover by breaking constraint 
«NSLayoutConstraint:0x102a82e0 H:[UILabel:0x102a8120(19)]»

Break on objc_exception_throw to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in «UIKit/UIView.h» may also be helpful.
2013-09-07 16:19:42.334 AgendaScolaire[13737:a0b] Unable to simultaneously satisfy constraints.

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x102a82e0 H:[UILabel:0x102a8120(19)]>

Break on objc_exception_throw to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
2013-09-07 16:19:42.334 AgendaScolaire[13737:a0b] Unable to simultaneously satisfy constraints.

**Comment by rboisjoly** [comment]
Ah, I was also setting the size of the subviews... conflict, there.

**Comment by smileyborg** [comment]
Ah, got it working then?

**Comment by rboisjoly** [comment]
got rid of the error... things not yet looking good, but no error is better :-) I'm surprised by how these constraints interact, not always what I expect.

I'll keep you posted.

Thanks for the amazingly fast response!

**Comment by smileyborg** [comment]
No worries :) Auto Layout can be quite difficult & frustrating, even with a nice interface like this haha.

Let me know if you run into any other issues! Good luck!
